File: competitive_programming/leetcode/07_typescript/242_valid_anagram.ts
function validAnagram(s: string, t: string): boolean {
  return s.split("").sort().join("") === t.split("").sort().join("");
}

if (validAnagram("anagram", "nagaram") !== true || validAnagram("rat", "car") !== false) throw new Error("valid anagram failed");
console.log("242 valid anagram ok");

File: competitive_programming/leetcode/11_golang/242_valid_anagram.go
package main

import (
	"fmt"
	"strings"
	"sort"
)

func validAnagram(s, t string) bool {
	left := strings.Split(s, "")
	right := strings.Split(t, "")
	sort.Strings(left)
	sort.Strings(right)
	return strings.Join(left, "") == strings.Join(right, "")
}

func main() {
	if !validAnagram("anagram", "nagaram") || validAnagram("rat", "car") {
		panic("valid anagram failed")
	}
	fmt.Println("242 valid anagram ok")
}
